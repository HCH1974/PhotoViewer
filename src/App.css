#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

#photo-viewer-title{
  size:0.05em;
  padding: 0.5rem;
  text-align: left;
  margin: 0;
}

#image-selector-title{
  size:0.05em;
  padding: 0.5rem;
  text-align:left;
  margin: 0;
}

#whole-page {

display: grid;
grid-column: 40%, 60% ;

}

#main-photo {

  grid-column-start: 1;
  grid-column-end: 1;

}

#photo-grid {

  grid-column-start: 2;
  grid-column-end: 2;
  
}