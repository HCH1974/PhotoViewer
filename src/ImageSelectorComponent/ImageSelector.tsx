import React from "react";  // import React (to provide access to TSX)
import "./ImageSelector.css";

const brokenImages = [1, 24, 32, 36, 44, 47];

function getImageUrls() {

    const urls: string[] = [];

    for (let i = 0; i < 50; i++) {

        if (!brokenImages.includes(i)) {
            const imageNumberString = i.toString().padStart(2, '0');
            urls.push(`https://picsum.photos/id/6${imageNumberString}/1600/900.jpg`)
        }
    }

    return urls;
}

export {getImageUrls};

function ImageSelector(props: {
    selected: string;
    setSelected: React.Dispatch<React.SetStateAction<string>>}) {    // declare angd export new function called 'Image Selector'



    const imageUrls = getImageUrls();

    // const [image, setImage] = React.useState("https://picsum.photos/id/223/1600/900.jpg");

    const images = imageUrls.map(imageUrl => {
         return <img src={imageUrl} onClick={function() {props.setSelected(imageUrl)}} className="individualImage"/>
           })


return (                
    <div className="imageGallery"> 
         
      {images} 

    </div>
);

}


export { ImageSelector };