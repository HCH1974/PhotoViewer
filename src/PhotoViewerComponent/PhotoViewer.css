.displayPhoto {

width:80%;
height:50%;
left: 0;
    
}