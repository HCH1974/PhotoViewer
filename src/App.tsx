import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { PhotoViewer } from './PhotoViewerComponent/PhotoViewer';
import { ImageSelector } from './ImageSelectorComponent/ImageSelector';

function App() {

  const [selected, setSelected]= useState("https://picsum.photos/id/223/1600/900.jpg");


  return (
    <div id="whole-page">
      <div id="main-photo">
      <p id="photo-viewer-title">Hello world</p>
      <PhotoViewer  src={selected}/>
      </div>
      <div id="photo-grid">
      <p id="image-selector-title">Image Selector {selected}</p>
      <ImageSelector selected={selected} setSelected={setSelected} />
      </div>
    </div>
      
  );
}

export default App
