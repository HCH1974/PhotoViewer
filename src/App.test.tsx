import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { PhotoViewer } from './PhotoViewerComponent/PhotoViewer';
import { ImageSelector } from './ImageSelectorComponent/ImageSelector';
import { getImageUrls } from './ImageSelectorComponent/ImageSelector';
import renderer from 'react-test-renderer';
import { create } from 'react-test-renderer';

test('renders Hello world text', () => {
  const { getByText } = render(<App />);
  const textElement = getByText(/Hello world/i);
  expect(textElement).toBeInTheDocument();
});

test('checks the first image displays correctly in PhotoViewer component', () => {
  render(<PhotoViewer src="https://picsum.photos/id/223/1600/900.jpg" />)
  const testImage = document.querySelector("img") as HTMLImageElement;
  expect(testImage.src).toContain("https://picsum.photos/id/223/1600/900.jpg");
});

test('getImagesUrls function correctly contains the first expected image URL', () => {

  const imageUrls = getImageUrls();
  expect(imageUrls[0]).toContain("https://picsum.photos/id/600/1600/900.jpg");
})

test('getImagesUrls function does not include broken image', () => {

  const imageUrls = getImageUrls();
  expect(imageUrls).not.toContain("https://picsum.photos/id/601/1600/900.jpg");

})


describe("My Component", () => {
  it("Test that snapshot of PhotoViewer matches previous snapshot", async () => {
    const tree = create(<PhotoViewer src="https://picsum.photos/id/223/1600/900.jpg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});