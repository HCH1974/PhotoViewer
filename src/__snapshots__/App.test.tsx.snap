// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component Test that snapshot of PhotoViewer matches previous snapshot 1`] = `
<div>
  <img
    className="displayPhoto"
    src="https://picsum.photos/id/223/1600/900.jpg"
  />
</div>
`;
